src/app/FakeLib/TypeScript.fs

```diff
@@ -56,7 +55,14 @@ let TypeScriptDefaultParams =
       OutputPath = null
       ToolPath = 
             if isUnix then "tsc"
-            else findPath "TypeScriptPath" TypeScriptCompilerPath "tsc.exe"
+            else 
+                let paths = 
+                    [ System.Environment.GetFolderPath System.Environment.SpecialFolder.ProgramFiles; System.Environment.GetFolderPath System.Environment.SpecialFolder.ProgramFilesX86]
+                    |> List.map (fun p -> p </> TypeScriptCompilerPrefix)
+                    |> List.collect (fun p -> try System.IO.DirectoryInfo(p).GetDirectories() |> List.ofArray with | _ -> [])
+                    |> List.map (fun di -> di.FullName)
+                    |> List.sortDescending
```

Comment:
Sorting by name will only work for single-digit versions - this will pick 9.0 over 10.0 for example